plasmapy/plasma/grids.py

```diff
@@ -12,11 +12,28 @@
 import numpy as np
 import pandas as pd
 import scipy.interpolate as interp
+import warnings
 import xarray as xr
 
 from abc import ABC
 from typing import Union
 
+# These standard keys are used throughout PlasmaPy to refer to certain
+# plasma quantities. This dictionary also provides the expected unit.
+recognized_keys = {
+    "x": ("x spatial position", u.m),
+    "y": ("y spatial position", u.m),
+    "z": ("z spatial position", u.m),
+    "rho": ("Mass density", u.kg / u.m ** 3),
+    "E_x": ("Electric field (x component)", u.V / u.m),
+    "E_y": ("Electric field (y component)", u.V / u.m),
+    "E_z": ("Electric field (z component)", u.V / u.m),
+    "B_x": ("Magnetic field (x component)", u.T),
+    "B_y": ("Magnetic field (y component)", u.T),
+    "B_z": ("Magnetic field (z component)", u.T),
+    "phi": ("Electric Scalar Potential", u.V),
+}
+
```

Comment:
These are really specific to the grid objects, so it's probably best to add these to `AbstractGrid` as a class attribute.  Then they would be properly documented.  This would look something like...

```python
class AbstractGrid(ABC):
    """..."""

    # quantities recognized by the grid mechanics
    _recognized_quantities = {...}

    @property
    def recognized_quantities(self):
        """Give a description."""
        return self._recognized_quantities

class PlasmaGrid(AbstractGrid):
    """..."""

    _recognized_quantities = {
        **AbstractGrid._recognized_quantities,
        "n_i": ("ion number density", u.m ** -3),
    }
```

In this setup, `recognized_quantities` will be properly documented for `AbstractGrid` and all its sub-classes.  Each sub-class can override or extend the recognized quantities without breaking any of the documentation.

----

This is completely your preference, but implementing a named tuple could be helpful here.  This would look something like...

```python
from collections import namedtuple

RecognizedQuantites = namedtuple("RecognizedQuantities", ["name", "long_descr", "unit"])

# implement in recognized_keys
recognized_keys = [
    RecognizedQuantities("x", "x spacial position", u.m),
    ...,
    RecognizedQuantites("B_Z", "Magnetic field (z componenet)", u.T),
    ...,
]
```

Then,
```python
>>> recognized_keys[0].name == recognized_keys[0][0]
True
>>> recognized_keys[0].name
'x'
>>> recognized_keys[0].long_descr
'x spacial position'
>>> recognized_keys[0].unit
Unit("m")
```

Reply:
Ok, I'll move this into AbstractGrid for now. 

In the future I think there's a case for having a structure like this that's used broadly throughout PlasmaPy. Certainly it would be bad if different modules used different standard key names. I'll tag @namurphy and @StanczakDominik to also think about this possibility in the future.

Reply:
I've chosen to leave this as a regular dictionary for now because it's useful to be able to quickly access the entries by key. I'm constructing this dictionary automatically from the list format you suggested above to minimize repetition.